flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -0,0 +1,289 @@
+      this.readerFunction = newReaderFunction;
+      return this;
+    }
+
+    public Builder caseSensitive(boolean newCaseSensitive) {
+      this.contextBuilder.caseSensitive(newCaseSensitive);
+      return this;
+    }
+
+    public Builder useSnapshotId(Long newSnapshotId) {
+      this.contextBuilder.useSnapshotId(newSnapshotId);
+      return this;
+    }
+
+    public Builder streamingStartingStrategy(StreamingStartingStrategy newStartingStrategy) {
+      this.contextBuilder.startingStrategy(newStartingStrategy);
+      return this;
+    }
+
+    public Builder startSnapshotTimestamp(Long newStartSnapshotTimestamp) {
+      this.contextBuilder.startSnapshotTimestamp(newStartSnapshotTimestamp);
+      return this;
+    }
+
+    public Builder startSnapshotId(Long newStartSnapshotId) {
+      this.contextBuilder.startSnapshotId(newStartSnapshotId);
+      return this;
+    }
+
+    public Builder endSnapshotId(Long newEndSnapshotId) {
+      this.contextBuilder.endSnapshotId(newEndSnapshotId);
+      return this;
+    }
+
+    public Builder asOfTimestamp(Long newAsOfTimestamp) {
+      this.contextBuilder.asOfTimestamp(newAsOfTimestamp);
+      return this;
+    }
+
+    public Builder splitSize(Long newSplitSize) {
+      this.contextBuilder.splitSize(newSplitSize);
+      return this;
+    }
+
+    public Builder splitLookback(Integer newSplitLookback) {
+      this.contextBuilder.splitLookback(newSplitLookback);
+      return this;
+    }
+
+    public Builder splitOpenFileCost(Long newSplitOpenFileCost) {
+      this.contextBuilder.splitOpenFileCost(newSplitOpenFileCost);
+      return this;
+    }
+
+    public Builder streaming(boolean streaming) {
+      this.contextBuilder.streaming(streaming);
+      return this;
+    }
+
+    public Builder monitorInterval(Duration newMonitorInterval) {
+      this.contextBuilder.monitorInterval(newMonitorInterval);
+      return this;
+    }
+
+    public Builder nameMapping(String newNameMapping) {
+      this.contextBuilder.nameMapping(newNameMapping);
+      return this;
+    }
+
+    public Builder project(Schema newProjectedSchema) {
+      this.contextBuilder.project(newProjectedSchema);
+      return this;
+    }
+
+    public Builder filters(List<Expression> newFilters) {
+      this.contextBuilder.filters(newFilters);
+      return this;
+    }
+
+    public Builder limit(long newLimit) {
+      this.contextBuilder.limit(newLimit);
+      return this;
+    }
+
+    public Builder includeColumnStats(boolean newIncludeColumnStats) {
+      this.contextBuilder.includeColumnStats(newIncludeColumnStats);
+      return this;
+    }
+
+    public Builder planParallelism(int planParallelism) {
+      this.contextBuilder.planParallelism(planParallelism);
+      return this;
+    }
+
+    public Builder properties(Map<String, String> properties) {
+      contextBuilder.fromProperties(properties);
+      return this;
+    }
+
+    public IcebergSource<T> build() {
```

Comment:
You could use `public <T> IcebergSource<T> build()` here to allow a bit easier type customization.